ledger/participant-integration-api/src/main/scala/platform/indexer/ha/PollingChecker.scala

```diff
@@ -52,19 +52,21 @@ class PollingChecker(
   // complete pool.
   def check(): Unit = synchronized {
     logger.debug(s"Checking...")
-    Try(checkBody) match {
-      case _ if closed =>
-        throw new Exception(
-          "Internal Error: This check should not be called from outside by the time the PollingChecker is closed."
-        )
+    if (closed) {
```

Comment:
Agree that it is better not to execute the body if closed.

Reply:
@simonmaxen-da I think you were right, rewamped the logic there so it is more obvious what is happening.